File: lintcode/naive/366_fibonacci.py
'''
Link: https://www.lintcode.com/problem/fibonacci/description
'''

# There is a method using recursion with memoization, but I did not use it.
class Solution:
    """
    @param n: an integer
    @return: an ineger f(n)
    """
    def fibonacci(self, n):
        # write your code here
        third = 0
        second = 0
        first = 0
        if n == 1:
            return first
        if n >= 2:
            second += 1
            if n == 2:
                return second
        for _ in range(2, n):
            third = second + first
            first = second
            second = third
        return third

File: lintcode/naive/366_fibonacci.java
public class Solution {
    /**
     * @param n: an integer
     * @return: an ineger f(n)
     */
    public int fibonacci(int n) {
        // write your code here
        int first = 0;
        int second = 1; 
        int third = 0;
        if (n == 1)
            return first;
        if (n == 2){
            return second;
        }
        for (int i = 3; i <= n; i++){
            third = first + second;
            first = second;
            second = third;
        }
        return third;        
    }
}
